src/Symfony/Component/AssetMapper/ImportMap/Resolver/JsDelivrEsmResolver.php

```diff
@@ -72,6 +72,7 @@ public function resolvePackages(array $packagesToRequire): array
 
             $version = $response->toArray()['version'];
             $requiredPackages[$i][1] = $this->httpClient->request('GET', sprintf($this->distUrlPattern, $packageName, $version, $filePath));
+            $requiredPackages[$i][4] = $version;
```

Comment:
Test should be added if we agree to fetch version at "require" step.

Reply:
I like that you added `version` to `importMapEntry`. My thinking is that we should *not* set this when you require a package, because for all of our normal resolvers, the package+version pattern is perfectly predictable. I think we can only realistically have the problem where we can't find the version of someone added a package manually to importmap.php that points somewhere different. If the user has that situation, they can add the `version`. But otherwise, you will never see or be aware of this key.

To support this, we'll probably need to add a method to `PackageResolverInterface` - e.g. `getVersion(string $url): ?string`.

Reply:
I added a `getPackageVersion` to the `PackageResolverInterface` in order to retrieve the package version. It's called when a version is not explicitly defined in the `importmap.php` file.